
On illumos, sendfile(fd1, fd1) succeeds without raising an exception

diff -wpruN '--exclude=*.orig' a~/Lib/test/test_shutil.py a/Lib/test/test_shutil.py
--- a~/Lib/test/test_shutil.py	1970-01-01 00:00:00
+++ a/Lib/test/test_shutil.py	1970-01-01 00:00:00
@@ -2396,7 +2396,7 @@ class _ZeroCopyFileTest(object):
     def test_same_file(self):
         self.addCleanup(self.reset)
         with self.get_files() as (src, dst):
-            with self.assertRaises(Exception):
+            #with self.assertRaises(Exception):
                 self.zerocopy_fun(src, src)
         # Make sure src file is not corrupted.
         self.assertEqual(read_file(TESTFN, binary=True), self.FILEDATA)
